Assets/Scripts/System/StateMachine/States/AttractState.cs

```diff
@@ -6,54 +6,147 @@
+
+
+    //STATE BASE FUNCTIONS
 
     override public void OnStateEnter(Animator animator, AnimatorStateInfo info, int layerIndex)
     {
+        //Call the inherited functions to get the helper variable
         base.OnStateEnter(animator, info, layerIndex);
+        
+        helper.attract.SetActive(true);
 
-        numberOfItems = GM.Instance.data.attractFiles.Count;
-        currentItem = 0;
+        numberOfItems = GM.Instance.data.attractItems.Count;
 
-        if (numberOfItems <= 0) {
+
+        if (numberOfItems <= 0)
+        {
             GM.Instance.logger.Warn("No valid files found in Attract folder (" + GM.Instance.options.attractPath + ")");
             animator.SetTrigger("NextState");
-        } else {
-            PlayVideo(0);
+        }
+        else
+        {
+            ShowCurrentItem();
         }
     }
 
     override public void OnStateExit(Animator animator, AnimatorStateInfo info, int layerIndex)
     {
-
+        helper.attract.SetActive(false);
     }
 
     override public void OnStateUpdate(Animator animator, AnimatorStateInfo info, int layerIndex)
     {
         if (Input.anyKeyDown)
             animator.SetTrigger("NextState");
 
-        playDelay -= Time.deltaTime;
+        displayTime -= Time.deltaTime;
+
+        //Do stuff when the delay is over
+        if (displayTime < 0)
+        {
+            //When playing video, only switch if video is done
+            if (GetCurrentItem().type == AttractItem.AttractItemType.Video)
+            {
+                if (!GM.Instance.video.player.isPlaying)
+                    ShowNextItem();
+            }
 
-        if (playDelay < 0 && !GM.Instance.video.player.isPlaying)
-            PlayNextVideo();
+            //On text or image, just go to the next one
+            else
+            {
+                ShowNextItem();
+            }
+        }
     }
 
-    private void PlayVideo(int number)
+
+    //STATE FUNCTIONS
+
+    /// <summary>
+    /// This shows the current attract item
+    /// </summary>
+    private void ShowCurrentItem()
     {
-        playDelay = 1;
-        GM.Instance.video.PlayVideo(GM.Instance.data.attractFiles[number], false, true);
+        if (GetCurrentItem() != null)
+        {
+            switch (GetCurrentItem().type)
+            {
+                case AttractItem.AttractItemType.Image:
+                    DisplayImage();
+                    return;
+
+                case AttractItem.AttractItemType.Text:
+                    DisplayText();
+                    return;
+
+                case AttractItem.AttractItemType.Video:
+                    DisplayVideo();
+                    return;
+            }
+        }
+
+        else
+        {
+            GM.Instance.logger.Debug("Attract: Item Number does not exist.");
+        }
     }
 
-    private void PlayNextVideo()
+    private void ShowNextItem()
     {
         currentItem++;
 
         if (currentItem > numberOfItems - 1)
             currentItem = 0;
```

Comment:
Here's a trick. Rather than increment and then check if you went too far, you can just mod the sucka:

```c#
currentItem = (currentItem + 1) % numberOfItems;
```

Not that it actually matters. :joy:

Reply:
ooo nice!  I like this!